supervisor/docker/network.py

```diff
@@ -42,55 +68,74 @@ def containers(self) -> list[str]:
     @property
     def gateway(self) -> IPv4Address:
         """Return gateway of the network."""
-        return DOCKER_NETWORK_MASK[1]
+        return DOCKER_IPV4_NETWORK_MASK[1]
 
     @property
     def supervisor(self) -> IPv4Address:
         """Return supervisor of the network."""
-        return DOCKER_NETWORK_MASK[2]
+        return DOCKER_IPV4_NETWORK_MASK[2]
 
     @property
     def dns(self) -> IPv4Address:
         """Return dns of the network."""
-        return DOCKER_NETWORK_MASK[3]
+        return DOCKER_IPV4_NETWORK_MASK[3]
 
     @property
     def audio(self) -> IPv4Address:
         """Return audio of the network."""
-        return DOCKER_NETWORK_MASK[4]
+        return DOCKER_IPV4_NETWORK_MASK[4]
 
     @property
     def cli(self) -> IPv4Address:
         """Return cli of the network."""
-        return DOCKER_NETWORK_MASK[5]
+        return DOCKER_IPV4_NETWORK_MASK[5]
 
     @property
     def observer(self) -> IPv4Address:
         """Return observer of the network."""
-        return DOCKER_NETWORK_MASK[6]
+        return DOCKER_IPV4_NETWORK_MASK[6]
+
+    def _get(self, network_id: str) -> docker.models.networks.Network:
+        """Get a network by ID."""
+        return self.docker.networks.get(network_id)
 
-    def _get_network(self) -> docker.models.networks.Network:
+    def _create(self, **kwargs: Any) -> docker.models.networks.Network:
+        """Create a new network."""
+        return self.docker.networks.create(**kwargs)
+
+    def _get_network(self, enable_ipv6: bool = False) -> docker.models.networks.Network:
         """Get supervisor network."""
         try:
-            return self.docker.networks.get(DOCKER_NETWORK)
+            if network := self._get(DOCKER_NETWORK):
+                if network.attrs.get(DOCKER_ENABLEIPV6) == enable_ipv6:
+                    return network
+                _LOGGER.info(
+                    "Migrating Supervisor network to %s",
+                    "IPv4/IPv6 Dual-Stack" if enable_ipv6 else "IPv4-Only",
+                )
+                if containers := network.containers:
+                    if all(
+                        container.name in ("hassio_supervisor", "hassio_observer")
```

Comment:
There is a constant for the observer `OBSERVER_DOCKER_NAME`. I don't think there is one for Supervisor, but I guess we should introduce one.

```suggestion
                        container.name in ("hassio_supervisor", OBSERVER_DOCKER_NAME)
```

We must reattach the Observer after recreating the network. The Supervisor gets attached later anyways, but probably easier to just reattach both.

Reply:
I was looking for constants, but I started (and ended up with the supervisor) so I thought the observer proly didn't have any either. 😆

Reply:
I'm not sure where to find the aliases and IPs for reattaching back... 🤔